base/strings/util.jl

```diff
@@ -499,29 +453,38 @@ julia> s = b"01abEF"
  0x45
  0x46
 
-julia> d = zeros(UInt8, 3)
-3-element Array{UInt8,1}:
- 0x00
- 0x00
- 0x00
-
-julia> hex2bytes!(d, s)
+julia> hex2bytes(a)
 3-element Array{UInt8,1}:
  0x01
  0xab
  0xef
 ```
 """
+function hex2bytes end
+
+hex2bytes(s::AbstractString) = hex2bytes(Vector{UInt8}(String(s)))
```

Comment:
This makes an additional copy of the data, but is still faster than the old implementation on my machine because it operates on bytes rather than unicode characters.  More importantly, it avoids code duplication.

(In any case, it's not clear to me how performance-critical this function is.)